File: check-db.js
const { Pool } = require('pg');
require('dotenv').config({ path: '.env.local' });

async function checkSchema() {
  const pool = new Pool({
    connectionString: process.env.DATABASE_URL,
  });
  
  try {
    const res = await pool.query(`
      SELECT column_name, data_type 
      FROM information_schema.columns 
      WHERE table_name = 'posts'
    `);
    console.log('Posts Schema:', res.rows);
    
    const constraints = await pool.query(`
      SELECT conname, pg_get_constraintdef(c.oid)
      FROM pg_constraint c
      JOIN pg_namespace n ON n.oid = c.connamespace
      WHERE conrelid = 'posts'::regclass
    `);
    console.log('Posts Constraints:', constraints.rows);

  } catch (e) {
    console.error('Schema check failed:', e);
  } finally {
    await pool.end();
  }
}

checkSchema();

File: pipeline/check_db.py
from pipeline.db import db
import json

def check_table(table):
    try:
        cols = db.fetch_all(f"SELECT column_name, data_type FROM information_schema.columns WHERE table_name = '{table}'")
        print(f"\n--- {table} ---")
        for c in cols:
            print(f"{c['column_name']} ({c['data_type']})")
    except Exception as e:
        print(f"Error checking {table}: {e}")

check_table('posts')
check_table('debates')
check_table('agents')
check_table('rss_feeds')
